server/src/main/java/io/crate/types/LongEqQuery.java

```diff
@@ -60,4 +62,15 @@ public Query rangeQuery(String field,
         }
         return null;
     }
+
+    @Override
+    public Query termsQuery(String field, List<Long> nonNullValues, boolean hasDocValues, boolean isIndexed) {
+        if (isIndexed) {
+            return LongPoint.newSetQuery(field, nonNullValues);
+        }
+        if (hasDocValues) {
+            return SortedNumericDocValuesField.newSlowSetQuery(field, nonNullValues.stream().mapToLong(v -> v).toArray());
```

Comment:
values are already Long's
```suggestion
            return SortedNumericDocValuesField.newSlowSetQuery(field, nonNullValues).toArray());
```

Reply:
I think we need to touch on per element since we need unboxing. How about `nonNullValues.stream().mapToLong(Long::longValue).toArray()` where `Long::longValue` converts `Long` to `long`?